flink-libraries/flink-table/src/main/scala/org/apache/flink/table/descriptors/SchemaValidator.scala

```diff
@@ -148,6 +148,13 @@ object SchemaValidator {
 
     val schema = properties.getTableSchema(SCHEMA)
 
+    // add all source fields first because rowtime might reference one of them
+    toScala(sourceSchema).map(_.getColumnNames).foreach { names =>
```

Comment:
Hi @twalthr, can we check the used `TimestampExtractor` here? Specifically, if it's an `ExistingField`, we only included the target fields; if it's a `StreamRecordTimestamp` we don't include extra fields; and only if it's a custom extractor we include all the source fields.

Reply:
Yes @xccui, we could change the logic like this. Do you have cases where we need such behavior?

Reply:
Not really. I was justing refactoring #5610. For convenience, I used the existing class `org.apache.flink.formats.avro.generated.User` in a test case, but it gets so many fields to be mapped. 😄

Reply:
I think we should first remove the added source fields before adding the explicit mappings with the following snippet. 
```
// add explicit mapping
case Some(source) =>
    // should add mapping.remove(source)
    mapping.put(name, source)
```